File: lab0-srv/src/main/java/pt/um/tf/lab0/lab0srv/Account.java
package pt.um.tf.lab0.lab0srv;

import java.util.concurrent.atomic.AtomicInteger;

/**
 * Account is just an integer based balance.
 */
public class Account {
    private int balance  = 0;

    protected synchronized boolean movement(int mov)  {
        boolean res = true;
        if (mov > 0) {
            balance += mov;
        }
        else {
            if (-mov > balance) {
                res = false;
            }
            else {
                balance += mov;
            }
        }
        return res;
    }

    public int getBalance() {
        return balance;
    }
}

File: lab0-srv/src/main/kotlin/pt/um/tf/lab0/lab0srv/account.kt
package pt.um.tf.lab0.lab0srv

class Account {
    var balance : Int = 0

    fun movement(mov : Int): Boolean {
        var res = true
        if (mov > 0) {
            balance += mov
        }
        else {
            if (-mov > balance) {
                res = false
            }
            else {
                balance -= -mov
            }
        }
        return res
    }
}